gprofiler/profilers/java.py

```diff
@@ -54,6 +55,23 @@ def get_ap_log(self) -> str:
         return self._ap_log
 
 
+class AsyncProfiledProcessMonitor:
+    def __init__(self):
+        self._attached_processes = []
+        self._proc_events_listener = ProcEventsListener()
+        self._proc_events_listener.start()
+        self._proc_events_listener.register_exit_callback(self._proc_exit_callback)
+
+    def _proc_exit_callback(self, pid, exit_code):
+        if pid in self._attached_processes:
+            if exit_code != 0:
```

Comment:
We want to check the exit_signal here (JVM dies with SIGABRT I think, or SIGKILL for OOM, or SIGTERM, ...)

Reply:
Talked IRL - it's complicated. @YishaiZinkin , I suggest you spend more time looking into after *after* moving the existing mechanism to the utils library. Currently, this won't work for the SIGKILL case you've shown me (because the main thread seemed to exit with `0`, and the main thread is the PID we're looking at...)

Reply:
The SIGKILL thing was a faux. And we decided to go with `!= 0` to begin with.